### ISSUE TITLE
Database is accessed during app config

### ISSUE THREAD

**Issue by benjaoming** [learningequality/morango#48]
Morango is currently hitting the database during `django.setup()` which is a problem for the database consistency check and database recovery (will never be able to use the Kolibri `dbrestore` command).

```
Traceback (most recent call last):
  File ".virtualenvs/kolibri3/bin/kolibri", line 9, in <module>
    load_entry_point('kolibri', 'console_scripts', 'kolibri')()
  File "/media/archive/code/kolibri/kolibri/utils/cli.py", line 592, in main
    initialize(debug=debug)
  File "/media/archive/code/kolibri/kolibri/utils/cli.py", line 160, in initialize
    django.setup()
  File ".virtualenvs/kolibri3/lib/python3.5/site-packages/django/__init__.py", line 27, in setup
    apps.populate(settings.INSTALLED_APPS)
  File ".virtualenvs/kolibri3/lib/python3.5/site-packages/django/apps/registry.py", line 116, in populate
    app_config.ready()
  File ".virtualenvs/kolibri3/lib/python3.5/site-packages/morango/apps.py", line 27, in ready
    InstanceIDModel.get_or_create_current_instance()
  File ".virtualenvs/kolibri3/lib/python3.5/site-packages/morango/models.py", line 122, in get_or_create_current_instance
    InstanceIDModel.objects.filter(current=True).update(current=False)
```

**Comment by jamalex** [comment]
@benjaoming Do you have any recommendations for where we should do bootstrapping like this instead?

**Comment by benjaoming** [comment]
It sounds like it's not a data migration.. because you want to ensure this state at all times? Or *is* it a data mgiration?

If not, this could be a more deliberate action. Does it belong in Morango or is it actually the calling library's responsibility to initialize Morango at the appropriate time?

One place for the caller (Kolibri) to initialize, would be in `kolibri.utils.cli.initialize` - this is not called when running `kolibri manage <command>`.

**Comment by ralphiee22** [comment]
hey @benjaoming, its not a data migration.
We are able to remove one of the calls to the database in apps.py, but we are also doing a `call_command('loaddata', 'scopedefinitions')` which loads fixtures into the database, which is relevant for getting syncing to work. so this seems like it belongs in morango? Way it works right now is fixtures are declared in kolibri, loaded through morango. wondering your thoughts on this?

**Comment by jamalex** [comment]
>Way it works right now is fixtures are declared in kolibri, loaded through morango

We in any case need to leave the fixtures defined in Kolibri, since they're application-specific (facilitydata, etc).

In that case, probably would make sense to shift the responsibility of loading the fixture into Kolibri itself too, so that it can choose the appropriate place and time to do it.

**Comment by ralphiee22** [comment]
The matching changes have been merged to 0.9.x which should be propagated to 0.10.
I believe I have removed all the code that hits the db in the `ready` method. All we do in that method is some attribute checking based on the `_meta` of the django class models.

**Comment by benjaoming** [comment]
@ralphiee22 cool! thanks! just looked in the current 0.3.1 release, and yes seems to be gone!

Closing this!
